### ISSUE TITLE
[Community Consultation] How can we reward Exercism's contributors?

### ISSUE THREAD

**Issue by iHiD** [exercism/exercism#4304]
There was an interesting discussion on the Team Slack the other day about how we can reward people for mentoring. In my mind, the same question also applies to those who maintain tracks or contribute in other ways. I am acutely aware of the dangers of interfering with the strong intrinsic motivation of helping others by adding external motivations, but I'm also interested in whether people have feelings as to how Exercism could give something back to them. 

If you have ideas on how we can give back to you for helping, pop a comment below. I'll collate them here at the top.

I don't want this issue to turn into a big debate on specific ideas. I really want to hear a series of individual's ideas and opinions. The Product Team will evaluate then take all the ideas and opinions on them and devise a strategy if we feel it's appropriate (we also reserve the right to do nothing if we think that's best). Feel free to edit your comments over time to comment on new ideas and please be respectful of the ideas in your language. Thank you :)

### Ideas
- Boosting contributors up the mentoring queue
- Providing nice showcase "badges" for profiles that can be used as part of CVs
- Having exclusive team t-shirts available to purchase (at cost) for contributors
- Get mentoring or career advice from someone experienced in the Exercism community.
- Have a GitHub organization that all mentors can join, so that when somebody sees my GitHub profile, they know immediately that's I'm an Exercism mentor.
-Maybe there could be a "The Mentor of the Week", or "Weekly Top 3", where for every language, you would proudly display mentors that have the highest score of solutions mentored this week * average score per mentoring.
- Exclusive Stickers
- Link the exercism contributions to github contributions?

**Comment by iHiD** [comment]
Two that were suggested in Slack:
- Boosting contributors up the mentoring queue
- Providing nice showcase "badges" for profiles that can be used as part of CVs

**Comment by kaiomagalhaes** [comment]
`Providing nice showcase "badges" for profiles that can be used as part of CVs`

Sounds like a good idea

**Comment by angelikatyborska** [comment]
**Intro**
I do the mentoring for two reasons:
1. Mentoring others is a great way to learn yourself.
2. For the warm feeling of having been helpful to somebody and for the "glory" of volunteering.

Thus I would want perks that improve those two aspects.

**Comments on already proposed ideas**

> Boosting contributors up the mentoring queue

I don't want that. I see the danger here of people signing up to be mentor *just* to get priority and not do any mentoring, or if this feature would check if you actually mentor (eg. mentor 3 exercises get priority in 1), it would encourage to do quantity over quality mentoring.

> Having exclusive team t-shirts available to purchase (at cost) for contributors

Yes, I would buy one! +1 to "glory" :).

> Providing nice showcase "badges" for profiles that can be used as part of CVs

I don't think I would put this on an actual CV, but possibly somewhere on my blog?

**My ideas**
- Have a GitHub organization that all mentors can join, so that when somebody sees my GitHub profile, they know immediately that's I'm an Exercism mentor. Maybe it would be possible to allow members of this organization to merge PRs for the mentoring notes? I'm not sure, that could lead to chaos. 
- Introduce some competition between mentors. Now, this one might be dangerous so it would need to be done very well. I know you collect this data about students rating mentors, and of course about how many exercises a mentor mentored. Maybe there could be a "The Mentor of the Week", or "Weekly Top 3", where for every language, you would proudly display mentors that have the highest score of solutions mentored this week * average score per mentoring.

**Comment by guygastineau** [comment]
I concur that boosting mentors in their language queues carries some danger.

Some of those dangers actually seem to disappear when applied to contributors; there are always tasks of varying levels that need doing.

Still, I'm not sure I want boosting. I certainly would like to get faster mentoring responses, but that isn't why I help maintain and mentor.

Reasons why I contribute and mentor:
1. I learn more deeply and rapidly through both activities.
2. I get to see people reach aha moments, and sometimes I get them in the process.
3. I get to be part of a great team :smile:
4. I get to help make this wonderful platform better for others, and for my future self.
----

Because of my reasons for participating above I think the t-shirts are the most appealing incentive.

I also think the badges would be a cool nerd trophy ;)

Actually being given priority in queues would feel like it is perverting the wonderful benefits we already receive for our hard work.

I understand that we are still looking to increase the numbers if truly active mentors and maintainers, but I think it is very important to remember that we are a community goals and ideals.  Thus, I think we should attempt to provide only those incentives that will not foster undesirable community or principles.

**Comment by luisalima** [comment]
I agree with @guygastineau. One thing that would be cool would be to somehow link the exercism contributions to github contributions? When we're contributing to exercism, as cool as it is, we are "removing" time from other potential OS projects. But I have no idea how that could be accomplished.

**Comment by F3PiX** [comment]
I 100% agree with the reasons for mentoring mentioned above.
Boosting: we already do have kind of boosting for mentors who want to add mentoring notes: at least in the Ruby channel, we have developed a habit that a mentor post their solution in the track channel to discuss 'possible solutions'.
Maybe we can make that kind of boosting more official/promote it to other channels, so mentors who make other mentors' life easier get a boost. But for that exercise only.
The same goes for new mentors or mentors new to an exercise; if someone asks for mentoring in the track channel, because they want to understand an exercise in order to mentor it, in the Ruby channel that is responded to with fast mentoring. This is also something that could be more 'official'.
I'm not in favour of boosting mentors just because they are mentors.

I was thinking about incentives that would make mentor's life easier, like the Mentor Notes, and the only other thing I came up with, is asking TextExpander to offer a free Team trial for some (extended) time. We could have team snippets then, as examples; in my experience, building a useful bunch of snippets is time consuming. (And, nice for TextExpander, I'm definitively going to buy it, now that I have a nice collection of snippets to work with.) 

As for badges, We do have the mentors pages, and they are public, right? How about repeating the 'has mentored x solutions and helped y students' info per track, with the mentor info, if even under an 'I' button. Maybe also a mentor's track progression bar. 

TL;DR Not in favour of boosting just because we're mentors. Would love to reward mentors with tools that makes mentoring easier.

**Comment by ptillemans** [comment]
Vinyl stickers are nice... no-one mentions stickers.... ;-)

Seriously all aspects should be considered. I would indeed not boost mentors just because they are mentors, on the other hand do you want experienced people which give good support to students relatively quickly through the core excercises so they can expand their scope gradually. I see that many mentors (me included) are quite reticent to start mentoring challenges they have not tried. So a modest unofficial boost based on student ratings and number of interactions might accomplish that and let the grapevine do the communicating. If it is sufficiently modest, no-one will notice who does not care, and those who do care will hear it from the grapevine and start noticing it.

**Comment by Stargator** [comment]
* Get mentoring or career advice from someone experienced in the Exercism community.

That could be helpful for folks who want to learn more about different aspects of software development or specific cases like making enterprise level Java-based backends.

**Comment by kzap** [comment]
- I do not agree with boosting as it can be gamed and abused. I think just more information would be better and thus mentors can choose if they want to specifically mentor other mentors:
    - perhaps some indicator on the list to show that a solution is from another mentor 
    - a dropdown to show solutions only from other mentor's
    - when viewing the solution of another mentor you should also be able to see some of their mentoring stats like when they joined and how many students/solutions they have mentored. This would allow someone to judge whether or not someone has a reputation as a mentor or is just starting and thus leave it up to the mentor to decide.

- As for rewards, I joined to help out the PHP track since there were few mentors, but maybe some way to show how much I contribute to Exercism, either in stats or ratings from others. It would be good social proof if you are a mentor and have high ratings from those you mentor. Ofcourse there must be safeguards put in place eventually to watch out for those who game the system, like ask for votes from people they mentor etc.... :)

**Comment by sfairchild** [comment]
I like the idea of an exclusive shop for mentors to get swag to show off being a mentor (stickers, t-shirts, etc). Maybe even just a regular shop with basic swag to help support the cost of running the community with an exclusive section for those that help out.

>  like ask for votes from people they mentor etc.... :)

I like that idea as an eventual feature. Allow the students to decide if their mentors comment was helpful of not, but I am not sure how that can be implemented to be a fair rating of a mentor. I could see situations where students rate their mentor low because they feel like their mentor was criticizing their work, while in reality the mentor was just trying to give alternative ways to help improve their answer (I have been guilty of feeling like this even knowing someone was trying to help, I think it is just part of human instinct).

**Comment by yawpitch** [comment]
I would run counter to some of the above and say I definitely do support the idea of boosting a mentor's work in other languages, though I would limit this to mentors who:

1. have at least met their quota for some number of weeks, and continue to do so AND
2. while meeting that quota their star-rating from students has been consistently high AND
3. the boost applies only to 1 core exercise at a time in 1 track they're an active student in

Potentially additional boosts could be awarded in tiers based on additional quality throughput or for sustained throughput over longer time spans. This would be inherently limiting, though, as at some point every mentor prone to the sort of moral turpitude worried about would have completed every core exercise in every language and would lose the benefits of their ill-gotten status.

Yes, the potential for gaming exists, but that's sort of the point of gamification ... if a mentor provides a high throughput at high quality, _their motivation is absolutely irrelevant_ from the student's perspective, and the _student's_ outcome is indistinguishable from that achieved by a less altruistic form of intrinsic motivation.

Does that mean that any / many mentors would respond to the incentive and allow it to reduce their intrinsic motivation to do good for others or to learn to become better teachers, etc, all well described above? Of course not ... the boost wouldn't be a reward *IN ANY WAY COMMENSURATE* to the value of the amount of time and energy they're volunteering, it would just be an _extremely_ light grease on the wheels of their ability to reach sufficient ability to mentor in other languages than their home track(s).

**Comment by karelvancamp** [comment]
+1 For the stickers, just because i'd like one, and make great promo (people only less laptops than t-shirts). From practical / cost point of view, using sytem like reddit trophies could be more feasible.

**Comment by jwir3** [comment]
> Providing nice showcase "badges" for profiles that can be used as part of CVs

Could consider issuing open badges (www.openbadges.org), similar to what Code School does. It's kind of nice for those that use it.

**Comment by chriswmercer** [comment]
Hello all, I was asked to post my thoughts from Slack -

> Boosting up the queue
I mean, maybe (( am always waiting for stuff to be approved) but maybe instead identifying who are mentors on other tracks so that I, as a mentor, can decide if I want to prioritise other mentors.

In addition - would it be worth having a mentor "team" for those mentors who want to help mentor the submissions of other mentors?

> Providing badges
yes good idea

> Exclusive T Shirts
good idea, I'd be down for that

> A mentor for mentors
good idea, I'll happily volunteer for helping with that

> GitHub org
good idea

> Mentor of the week
mmm maybe but I think it runs the risk of attracting the wrong sort to mentoring. It shouldn't be a competition.

> Stickers
great idea, I'll take a load :smile:

**Comment by ErikSchierboom** [comment]
Hello 👋

With the launch of Exercism v3, we are **closing all issues in this repository** to help give us a clean slate to detect new problems. If this issue is still relevant to Exercism v3 (e.g. it's a feature that we haven't implemented in v3, or a bug that still exists), please reopen it and we will review it and post an update on it as soon as we get chance.

Thanks for helping make Exercism better, and we hope you enjoy v3 🙂
